pkg/controller/bootstrap/bootstrapsigner_test.go

```diff
@@ -0,0 +1,172 @@
+/*
+Copyright 2014 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package bootstrap
+
+import (
+	"testing"
+
+	"github.com/davecgh/go-spew/spew"
+
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/unversioned"
+	"k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset/fake"
+	"k8s.io/kubernetes/pkg/client/testing/core"
+)
+
+func init() {
+	spew.Config.DisableMethods = true
+}
+
+func newBootstrapSigner() (*BootstrapSigner, *fake.Clientset) {
+	options := DefaultBootstrapSignerOptions()
+	cl := fake.NewSimpleClientset()
+	return NewBootstrapSigner(cl, options), cl
+}
+
+func newConfigMap(tokenID, signature string) *api.ConfigMap {
+	ret := &api.ConfigMap{
+		ObjectMeta: api.ObjectMeta{
+			Namespace:       api.NamespacePublic,
+			Name:            configMapClusterInfo,
+			ResourceVersion: "1",
+		},
+		Data: map[string]string{
+			kubeConfigKey: "payload",
+		},
+	}
+	if len(tokenID) > 0 {
+		ret.Data[signaturePrefix+tokenID] = signature
+	}
+	return ret
+}
+
+func newTokenSecret(tokenID, tokenSecret string) *api.Secret {
+	ret := &api.Secret{
+		ObjectMeta: api.ObjectMeta{
+			Namespace:       api.NamespaceSystem,
+			Name:            "token-" + tokenID,
```

Comment:
Not sure it matters but these are "bootstrap-token-" on my end, if so which should we sync on?